src/pack/index.ts

```diff
@@ -0,0 +1,43 @@
+import last from 'it-last'
+import pipe from 'it-pipe'
+
+import { CarWriter } from '@ipld/car'
+import { importer } from 'ipfs-unixfs-importer'
+import normalizeAddInput from 'ipfs-core-utils/src/files/normalise-input/index'
+import { ImportCandidateStream } from 'ipfs-core-types/src/utils'
+import { sha256 } from 'multiformats/hashes/sha2'
+
+import { Blockstore } from '../blockstore'
+import { MemoryBlockStore } from '../blockstore/memory'
+
+export async function pack ({ input, blockstore = new MemoryBlockStore() }: { input: ImportCandidateStream, blockstore: Blockstore }) {
```

Comment:
I think `pack` should require the calling code to pass in a block store, and then leave it up to the caller to destroy it afterwards if they want to